colpali_engine/trainer/contrastive_trainer.py

```diff
@@ -1,12 +1,180 @@
+from typing import Iterator, List, Optional
+
+import numpy as np
 import torch
-from transformers import Trainer
+from datasets import DatasetDict
+from torch.utils.data import BatchSampler, ConcatDataset, DataLoader, Dataset
+from transformers import Trainer, is_datasets_available
+from transformers.trainer_utils import seed_worker
+
+
+class SingleDatasetBatchSampler(BatchSampler):
```

Comment:
This was not the original goal of this PR.
I think it should be in a separate PR. But if you don't want to separate the integration of this feature, at least document the new feature in the PR discussion.